disttask/importinto/planner.go

```diff
@@ -397,6 +442,49 @@ func getSortedKVMetas(subTaskMetas [][]byte) (map[string]*external.SortedKVMeta,
 	return res, nil
 }
 
+func getSortedKVMetasOfMergeStep(subTaskMetas [][]byte) (map[string]*external.SortedKVMeta, error) {
+	result := make(map[string]*external.SortedKVMeta, len(subTaskMetas))
+	for _, subTaskMeta := range subTaskMetas {
+		var stepMeta MergeSortStepMeta
+		err := json.Unmarshal(subTaskMeta, &stepMeta)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		meta, ok := result[stepMeta.KVGroup]
+		if !ok {
+			result[stepMeta.KVGroup] = &stepMeta.SortedKVMeta
+			continue
+		}
+		meta.Merge(&stepMeta.SortedKVMeta)
+	}
+	return result, nil
+}
+
+func getSortedKVMetasForIngest(planCtx planner.PlanCtx) (map[string]*external.SortedKVMeta, error) {
+	kvMetasOfMergeSort, err := getSortedKVMetasOfMergeStep(planCtx.PreviousSubtaskMetas[StepMergeSort])
+	if err != nil {
+		return nil, err
+	}
+	kvMetasOfEncodeStep, err := getSortedKVMetasOfEncodeStep(planCtx.PreviousSubtaskMetas[StepEncodeAndSort])
```

Comment:
it means now we need the result of previous two steps in merge step？

Reply:
in `write&ingest` step, yes, as some kv-group might skip merge-sort